src/compiler/parser.ts

```diff
@@ -2655,10 +2650,11 @@ namespace ts {
                 case SyntaxKind.UndefinedKeyword:
                 case SyntaxKind.NeverKeyword:
                 case SyntaxKind.ObjectKeyword:
-                    // If these are followed by a dot, then parse these out as a dotted type reference instead.
-                    return tryParse(parseKeywordAndNoDot) || parseTypeReference();
                 case SyntaxKind.SymbolKeyword:
-                    return tryParse(parseSymbolType) || parseTypeReference();
+                // If these are followed by a dot, then parse these out as a dotted type reference instead.
+                    return tryParse(parseKeywordAndNoDot) || parseTypeReference();
+                case SyntaxKind.UniqueKeyword:

```

Comment:
Why not just add this as a case for `TypeOperator`, and make it an error for the operand to not be `symbol`? Should get better errors that way when people inevitably try `type Farenheight = unique number`.